data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/KyberApi.kt

```diff
@@ -0,0 +1,276 @@
+package com.vultisig.wallet.data.api.swapAggregators
+
+import com.vultisig.wallet.data.api.models.KyberSwapBuildRequest
+import com.vultisig.wallet.data.api.models.KyberSwapError
+import com.vultisig.wallet.data.api.models.KyberSwapErrorResponse
+import com.vultisig.wallet.data.api.models.KyberSwapRouteResponse
+import com.vultisig.wallet.data.api.models.KyberSwapToken
+import com.vultisig.wallet.data.api.models.KyberSwapTokensResponse
+import com.vultisig.wallet.data.api.models.quotes.KyberSwapQuoteDeserialized
+import com.vultisig.wallet.data.api.models.quotes.KyberSwapQuoteJson
+import com.vultisig.wallet.data.chains.helpers.EvmHelper
+import com.vultisig.wallet.data.models.Chain
+import com.vultisig.wallet.data.utils.KyberSwapQuoteResponseJsonSerializer
+import io.ktor.client.*
+import io.ktor.client.call.body
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.serialization.json.*
+import java.math.BigInteger
+import io.ktor.http.path
+import javax.inject.Inject
+import kotlin.text.lowercase
+
+interface KyberApi {
+
+    suspend fun getSwapQuote(
+        chain: Chain, srcTokenContractAddress: String, dstTokenContractAddress: String,
+        amount: String, srcAddress: String, isAffiliate: Boolean
+    ): KyberSwapQuoteDeserialized
+}
+
+class KyberApiImpl @Inject constructor(
+    private val httpClient: HttpClient,
+    private val kyberSwapQuoteResponseJsonSerializer: KyberSwapQuoteResponseJsonSerializer,
+    private val json: Json,
+) : KyberApi {
+    private val aggregatorApiBaseUrl = "https://aggregator-api.kyberswap.com"
+    private val tokenApiBaseUrl = "https://ks-setting.kyberswap.com"
+
+
+    override suspend fun getSwapQuote(
+        chain: Chain, srcTokenContractAddress: String, dstTokenContractAddress: String,
+        amount: String, srcAddress: String, isAffiliate: Boolean
+    ): KyberSwapQuoteDeserialized {
+        val sourceAddress =
+            if (srcTokenContractAddress.isEmpty()) NULL_ADDRESS else srcTokenContractAddress
+        val destinationAddress =
+            if (dstTokenContractAddress.isEmpty()) NULL_ADDRESS else dstTokenContractAddress
+
+        val response = httpClient.get(aggregatorApiBaseUrl) {
+            url {
+                path(
+                    chain.raw.lowercase(),
+                    "api/v1/routes",
+                )
+                parameters.append(
+                    "tokenIn",
+                    sourceAddress
+                )
+                parameters.append(
```

Comment:
Please group the with `apply`
url {
parameters.apply {
append("tokenIn", sourceAddress)
append("tokenOut", destinationAddress)
append("amountIn", amount)
append("saveGas", "false")
append("gasInclude", "true")
append("slippageTolerance", SLIPPAGE_TOLERANCE.toString())
append("isAffiliate", isAffiliate.toString())
parameter("sourceIdentifier", if (isAffiliate) CLIENT_ID else null)
parameter("referrerAddress", if (isAffiliate) REFERRER_ADDRESS else null)
}